### PR TITLE
Nmc 430-harmonizing of sharing permissions step 1

### PR DESCRIPTION
**Pull request by TSI-yogeshshejwadkar**
Folders
Internal Share:
we removed the checkboxes "Allow editing", "Allow creating" and "Allow deleting"
we added a radio button group containing the options: "Read only" (Nur Lesen), "Allow upload and editing" (Hochladen & Bearbeiten)
External Share: no changes
Link Share: no changes

Files
Internal Share:
we removed the checkbox "Allow editing"
we added a radio button group containing the options: "Read only" (Nur lesen), "Editing" (Bearbeiten)
External Share:
we removed the checkbox "Allow editing"
we added a radio button group containing the options: "Read only" (Nur lesen), "Editing" (Bearbeiten)
Link Share:
we removed the checkbox "Allow editing"
we added a radio button group containing the options: "Read only" (Nur lesen), "Editing" (Bearbeiten)

Signed-off-by: Yogesh Shejwadkar <yogesh.shejwadkar@t-systems.com>

### CODE REVIEW

**Comment by kesselb** [comment]
Hey :wave: 

it looks like a mistake, to me, that changes related to S3 and encryption are part of this pull request.

**Comment by juliushaertl** [comment]
In addition the commit is missing a `Signed-of-by` message as described in https://github.com/nextcloud/server/blob/master/.github/CONTRIBUTING.md#sign-your-work

**Comment by Salvador1313** [comment]
Это вы в этом знатоки. Я пока только на уровне физических символов
программирование придумывать начал.  До компьютерного тока недавно
добрался.  Смотрите сами как лучше будет....

вс, 29 авг. 2021 г., 16:46 TSI-yogeshshejwadkar ***@***.***>:

**Comment by juliushaertl** [comment]
@Salvador1313 Please stick to english for comments as we use that as language for communication on our github organization. :wink:

**Comment by Salvador1313** [comment]
Так?


вс, 29 авг. 2021 г., 16:46 TSI-yogeshshejwadkar ***@***.***>:

**Comment by TSI-yogeshshejwadkar** [comment]
> Hey 
> 
> it looks like a mistake, to me, that changes related to S3 and encryption are part of this pull request.

Removed the extra commits

**Comment by TSI-yogeshshejwadkar** [comment]
> In addition the commit is missing a `Signed-of-by` message as described in https://github.com/nextcloud/server/blob/master/.github/CONTRIBUTING.md#sign-your-work

Added the message

**Comment by kesselb** [comment]
| Before | After |
| --- | --- |
| Screenshot from 2021-08-30 22-09-19 | Screenshot from 2021-08-30 22-07-54 |
| Screenshot from 2021-08-30 22-17-42 | Screenshot from 2021-08-30 22-23-47 
| Screenshot from 2021-08-30 22-17-56 | Screenshot from 2021-08-30 22-24-03

**Comment by kesselb** [comment]
Well done :+1: 

image

To fix the CI: Build the changes locally (make build-js-production) and commit the above files.

**Comment by AndyScherzinger** [comment]
@juliushaertl @skjnldsv is this really v23 or does this need to be backported to v22, sames question for #28636 of course

**Comment by TSI-yogeshshejwadkar** [comment]
> Well done 
> 
> image
> 
> To fix the CI: Build the changes locally (make build-js-production) and commit the above files.

added the requested file. Thanks :)

**Comment by TSI-yogeshshejwadkar** [comment]
Updated conflict production build files

**Review by artonge — CHANGES_REQUESTED** [review]

File: apps/files_sharing/src/components/SharingEntry.vue [hunk 1]
```diff
@@ -44,34 +44,44 @@
...
+						@change="togglePermissions">
+						{{ t('files_sharing', 'Read only') }}
+					</ActionRadio>
+
+					<ActionRadio :checked="sharePermissions === publicUploadEValue"
+						:value="publicUploadEValue"
+						:disabled="saving"
+						:name="randomId"
+						@change="togglePermissions">
+						{{ t('files_sharing', 'Editing') }}
```
[the remark this answers is not in the document]
Comment (jancborchardt) [reply apps/files_sharing/src/components/SharingEntry.vue:82]: "Read and write" is very technical. @TSI-yogeshshejwadkar could you change the labels to "View only" and "Edit"?

**Comment by Leptopoda** [comment]
Just found this pr and want to add my opinion ;)
What happened to the upload only option?
Sometimes sharing a folder to a group and letting people just drop their stuff into it without fearing vandalism or accidental breakage.

nevertheless looks like a nice peace of ux improvement for many othervise confused users; so thanks!

File: apps/files_sharing/src/components/SharingEntryLink.vue [hunk 2]
```diff
@@ -178,12 +178,23 @@
...
+							@change="togglePermissions">
+							{{ t('files_sharing', 'Read only') }}
+						</ActionRadio>
+
+						<ActionRadio :checked="sharePermissions === publicUploadEValue"
+							:value="publicUploadEValue"
+							:disabled="saving"
+							:name="randomId"
+							@change="togglePermissions">
+							{{ t('files_sharing', 'Editing') }}
```
[the remark this answers is not in the document]
Comment (jancborchardt) [reply apps/files_sharing/src/components/SharingEntryLink.vue:195]: Wording-wise it should be "Edit" as per mockup: https://github.com/nextcloud/server/issues/26691
"Read and write" sounds way too technical.

File: apps/files_sharing/src/components/SharingEntry.vue [hunk 3]
```diff
@@ -44,34 +44,44 @@
...
-					:value="permissionsDelete"
-					:disabled="saving || !canSetDelete">
-					{{ t('files_sharing', 'Allow deleting') }}
-				</ActionCheckbox>
+					<ActionRadio :checked="sharePermissions === publicUploadRWValue"
+						:value="publicUploadRWValue"
+						:disabled="saving"
+						:name="randomId"
+						@change="togglePermissions">
+						{{ t('files_sharing', 'Allow upload and editing') }}
```
[the remark this answers is not in the document]
Comment (jancborchardt) [reply apps/files_sharing/src/components/SharingEntry.vue:63]: No, this is correct, the more human way to write it. "Read / write" is too technical.

**Pull request closed without merging**